ast/parser_ext.go

```diff
@@ -686,6 +715,17 @@ func setRuleModule(rule *Rule, module *Module) {
 	}
 }
 
+func setTermsJSONOptions(bodyTerms interface{}, jsonOptions *JSONOptions) {
```

Comment:
I agree, having a single function performing both `body[0].setJSONOptions(rhs.jsonOptions)` and `setTermsJSONOptions(body[0].Terms, ...` is better. I prefer not to move the first two instructions mentioned in the previous comment (i.e., `NewBody(...` and, `head.setJSONOptions(`).

The final result would be something like this:

```go
head.setJSONOptions(lhs.jsonOptions)

body := NewBody(NewExpr(BooleanTerm(true).SetLocation(rhs.Location)).SetLocation(rhs.Location))
setJSONOptions(body, &rhs.jsonOptions)
```

@johanfylling, is it ok if I commit this along the rebase with the current main branch?

Reply:
That'll be great, go for it 👍 
(I only meant for the `body[0].setJSONOptions(...)` to be moved, so we're completely on the same page 🙂 )

Reply:
Using the visitor further simplifies the logic, since the [Body](https://github.com/open-policy-agent/opa/blob/main/ast/visit.go#L325) case is handled alongside the [*Expr](https://github.com/open-policy-agent/opa/blob/main/ast/visit.go#L333) in [visit.go](https://github.com/open-policy-agent/opa/blob/main/ast/visit.go)